Installation.md

```diff
@@ -0,0 +1,259 @@
+## Setup ##
+
+### Notices before installation ###
+
+We assume you use [RVM](https://rvm.io/). If you don't, then replace `rvmsudo` with `sudo` during the installation process.
+
+:exclamation: RVM 1.17.x may have some [problems with rvmsudo](http://stackoverflow.com/questions/13765520/rvmsudo-command-not-working-properly).
+
+An X server is currently needed for browser tests to pass when using Selenium. `Xvfb` will do, but remember to set your `DISPLAY`.
+
+For testing environment PhantomJS (at least version 1.8.1) is required by Poltergeist gem to run headless browser tests. See [poltergeist](https://github.com/teampoltergeist/poltergeist) for documentation.
+
+If you are using Java 8, the following test will fail `RemoteSandboxForTesting should include tools.jar in the classpath for ant projects`. Please refer tp [issue #232](https://github.com/testmycode/tmc-server/issues/232) for further information.
+
+Below is a quick summary of the setup procedure. It is followed by detailed instructions for Ubuntu.
+
+### Quick summary ###
+
+The following programs should be installed first: `ruby` (and `bundler` gem), `PostgreSQL 9.2+`, `screen`,`git`, `zip`, `unzip`, `convert` (from ImageMagick), `javac`, `java`, `ant`, `mvn`, `gcc`, `make` and `bc`. Additionally, tmc-sandbox requires: `squashfs-tools`, `multistrap` `e2fsprogs`, `e2tools` and `build-essential`.
+
+1. Download submodules with `git submodule update --init --recursive`
+2. Install dependencies with `bundle install`
+3. Edit `config/site.yml`, based on `config/site.defaults.yml`.
+4. Setup PostgreSQL based on `config/database.yml` or configure `config/database.local.yml` according to your current postgres settings.
+5. Initialize rails database with `rake db:reset`
+6. Go to `ext/tmc-sandbox` and compile with `sudo make`. More information from [tmc-sandbox page](https://github.com/testmycode/tmc-sandbox).
+7. Go to `ext/tmc-sandbox/web` and install dependencies with `bundle install`. Compile extensions with `rake ext` and run tests with `rvmsudo rake test`.
+8. Compile the other stuff in `ext` by doing `rake compile`
+9. Install [tmc-check](https://github.com/testmycode/tmc-check) by running `rvmsudo make rubygems install clean` in `ext/tmc-sandbox/uml/output/tmc-check`.
+10. Run the test suite with `rvmsudo rake spec`.
+
+After you get the test suite to pass see [post-install instructions](#post-install-instructions).
+
+### Installation ###
+
+The following is a set of detailed instructions for setting tmc-server up on Ubuntu 14.04.
+
+#### Set Java, Ruby and PhantomJS versions to be installed ####
```

Comment:
Is this needed - anyway topic mention phantomjs and its not referenced in the body...